src/supervisor.sbvr

```diff
@@ -0,0 +1,461 @@
+
+	Fact type: application has api secret
+		Necessity: each application that has a slug that is equal to "balena-os/balena-supervisor" has at least one api secret.
+
+	Fact type: device belongs to application
+		Synonymous Form: application owns device
+		Necessity: each device belongs to at least one application.
+		-- Changed from "at most one application" to "at least" for multi-app -- valid?
+
+	Fact type: application has commit
+		Necessity: each application has exactly one commit.
+		Necessity: each commit belongs to exactly one application.
+		Necessity: each commit has a uuid that has a Length (Type) that is equal to 32.
+
+
+
+-- CONFIG / ENV VARS
+Term: value
+	Concept Type: Text (Type)
+
+Term: config var
+	Concept Type: Short Text (Type)
+
+	Fact type: config var has value
+		Necessity: each config var has exactly one value.
+
+	Fact type: application owns config var
+		Synonymous Form: config var belongs to application
+		Necessity: each application that has a config var must have a slug equal to "balena-os/balena-supervisor".
+		Necessity: each config var that belongs to an application must begin with "BALENA_" or "RESIN_".
+
+Term: env var
+	Concept Type: Short Text (Type)
+
+	Fact type: env var has value
+		Necessity: each env var has exactly one value.
+
+	Fact type: service owns env var
+		Synonymous Form: env var belongs to service
+
+
+
+-- UPDATE LOCK
+
+Term: update lock
+	Concept Type: Short Text (Type)
+	-- Path of update lockfile (Short Text (Type))?
+	-- Or integer written into lockfile (Integer (Type))?
+
+	Fact type: application has update lock
+		Necessity: each application has at most one update lock.
+
+	Fact type: service has update lock
+		Necessity: each service has at most one update lock.
+
+
+
+-- ACTION: TODO
+-- Since this is theoretical, this probably shouldn't initially be included in the model.	
+-- Eventually the SV or API could specify a timestamp in the future as part of the target state for the SV to execute an action
+
+Term: action
+	Concept Type: Date Time (Type)
+
+	Fact type: device implements action
+	-- Is "implements" a valid verb?? How can we translate this into the expectation of a POST/PATCH request in ODATA here?
+		Necessity: each device implements one of each action.
+
+	Term: blink
+	-- POST /v1/blink
+
+		Synonym: identify
+		Concept Type: action
+
+	Term: ping
+	-- GET /ping
+
+		Concept type: action
+
+	Term: purge
+	-- POST /v1/purge
+
+		Concept Type: action
+
+		Fact type: purge is force 
+
+	Term: reboot
+	-- POST /v1/reboot
+		Concept Type: action
+
+		Fact type: reboot is force
+
+	Term: regenerate api key
+	-- POST /v1/regenerate-api-key
+
+		Synonym: regenerate api secret
+		-- The current name is regenerate api key but "api key" is an ambiguous term since balenaAPI also has api keys.
+		-- We should use api secret which is the database table name as well as the name for the SV's api key in the open-balena-api model
+		Concept Type: action
```

Comment:
I haven't expanded more on the concept of scoped API secrets in this model, because I haven't had a chance to think about what a more ideal API key model could be for the Supervisor.

Reply:
Removed this section and anything else related to Supervisor API keys (API secret, not deviceApiKey) for the time being